src/test/java/com/teragrep/cfe_39/HdfsTest.java

```diff
@@ -0,0 +1,260 @@
+package com.teragrep.cfe_39;
+
+import com.teragrep.cfe_39.avro.SyslogRecord;
+import com.teragrep.cfe_39.consumers.kafka.HDFSWrite;
+import com.teragrep.cfe_39.consumers.kafka.KafkaController;
+import com.teragrep.cfe_39.consumers.kafka.RecordOffset;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.junit.jupiter.api.Assertions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+
+public class HdfsTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HdfsTest.class);
+
+    private static MiniDFSCluster hdfsCluster;
+    private static File baseDir;
+    private static Config config;
+
+    // Generate AVRO-files for testing the HDFS writes.
+    // @BeforeAll
+    public static void generateTestData() throws IOException, InterruptedException {
+        config = null;
+        try {
+            config = new Config();
+        } catch (IOException e){
+            LOGGER.error("Can't load config: " + e);
+            System.exit(1);
+        } catch (IllegalArgumentException e) {
+            LOGGER.error("Got invalid config: " + e);
+            System.exit(1);
+        }
+        startMiniCluster();
+        config.setMaximumFileSize(3000); // 10 loops (140 records) are in use at the moment, and that is sized at 36,102 bytes.
+        KafkaController kafkaController = new KafkaController(config);
+        kafkaController.run();
+    }
+
+    public static void startMiniCluster() throws IOException {
```

Comment:
Tests should not throw nor try/catch anything, use `Assertions.assertDoesNotThrow`, `Assertions.assertThrows`, `Assertions.assertAll` and such instead